Assignment-1/main.swift

```diff
@@ -0,0 +1,179 @@
+
+import Foundation
+
+class ItemDetails{
+     var name: String
+     var price: Double
+    var quantity: Int? = 1
+     var type: String
+    
+    init(_ name:String,_ price:Double,_ quantity:Int, _ type: String){
+        self.name = name
+        self.price = price
+        self.quantity = quantity
+        self.type = type
+    }
+    
+    func printItemDetail(){
+        print("Name of the item         \(name)")
+        print("Type of the item         \(type)")
+        print("Price of the item        \(price)")
+        print("Quantity of the item     \(quantity)")
+    }
+
+}
+
+// a protocol declaration for tax calculations
+protocol TaxCaculations{
+    func calculateTax()->(Double)
+    func printTax()
+}
+ 
+
+
+
+
+// array decleration for storing the details of the items.
+
+var rawItemList = [RawItem]()
+var manufacturedItemList = [ManufacturedItem]()
+var importedItemList = [ImportedItem]()
+
+
+
+func inputItems(){
+    
+    var price:Double?
+    var name:String?
+    var type:String?
+    var quantity:Int? = 1
+    
+    func isItCommand(inputString:String)-> (Bool){  // return true if current string is a command else returns false
+        if(inputString=="-name"){//
```

Comment:
Keep Strings in a Seperate String constant file

Reply:
okay